src/Chorus/UI/Clone/GetCloneFromInternetDialog.cs

```diff
@@ -261,11 +261,35 @@ private void _cancelButton_Click(object sender, EventArgs e)
 		}
 
 		private void OnDownloadClick(object sender, EventArgs e)
+		{
+			StartClone();
+		}
+
+		
+		/// <summary>
+		/// Starts a clone operation with the supplied information.
+		/// <param name="username">Username for Mercurial authentication</param>
+		/// <param name="password">Password for Mercurial authentication</param>
+		/// <param name="projectName">Name of the project to clone</param>
+		/// <param name="projectUri">URI where the project can be found</param>
+		/// </summary>
+		public void StartClone(string username, string password, string projectName, Uri projectUri)
```

Comment:
I think the public method would make more sense as a static method to

* create a `GetCloneFromInternetModel` from the arguments
* create a `GetCloneFromInternetDialog` from the model
* StartClone(); with no arguments